### PR TITLE
feat: Upgrade dog, clip to 0.6.7, Update CI (SC-7268)

### PR DESCRIPTION
**Pull request by lucas-manuel**
If anyone sees another opportunities to leverage 0.6.7 in these files (or the end) please let me know.

### CODE REVIEW

File: src/clip.sol [hunk 1]
```diff
@@ -55,13 +55,13 @@ contract Clipper {
...
-    address    public vow;   // Recipient of dai raised in auctions
-    VatLike    public vat;   // Core CDP Engine
-    DogLike    public dog;   // Liquidation module
-    SpotLike   public spot;  // Collateral price module
-    AbacusLike public calc;  // Current price calculator
+    address              public vow;   // Recipient of dai raised in auctions
+    VatLike    immutable public vat;   // Core CDP Engine
+    DogLike              public dog;   // Liquidation module
+    SpotLike   immutable public spot;  // Collateral price module
+    AbacusLike           public calc;  // Current price calculator
```
Comment (kmbarry1) [inline src/clip.sol:64]: Suggestion: separate `immutable` from mutable values for better organizatoin/clarity.

**Comment by kmbarry1** [comment]
In abaci.sol, there's an interface that should be uncommented (with corresponding inheritance tags and `override` keywords that should also be uncommented). If you were planning to do that in another PR, that's fine too.

**Comment by lucas-manuel** [comment]
@kmbarry1 I was thinking about adding that in another PR, but I think I'll just add it in this one.

**Review by kmbarry1 — APPROVED** [review]
LGTM

**Review by brianmcmichael — APPROVED** [review]

**Pull request merged**